GVRf/Sample/gvr-sample/src/org/gearvrf/sample/SampleActivity.java

```diff
@@ -13,17 +13,74 @@
  * limitations under the License.
  */
 
-
 package org.gearvrf.sample;
 
-import android.os.Bundle;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
 
 import org.gearvrf.GVRActivity;
+import org.gearvrf.GVRScreenshotCallback;
+import org.gearvrf.utility.Log;
+
+import android.graphics.Bitmap;
+import android.os.Bundle;
+import android.os.Environment;
+import android.view.MotionEvent;
 
 public class SampleActivity extends GVRActivity {
-    @Override
-    protected void onCreate(Bundle icicle) {
-        super.onCreate(icicle);
-        setScript(new SampleCubeScript(), "gvr_note4.xml");
-    }
+	@Override
+	protected void onCreate(Bundle icicle) {
+		super.onCreate(icicle);
+		setScript(new SampleCubeScript(), "gvr_note4.xml");
+	}
+
+	@Override
+	public boolean onTouchEvent(MotionEvent event) {
+		captureScreenLeft(newScreenshotCallback("screenshot_left.png"));
+		captureScreenRight(newScreenshotCallback("screenshot_right.png"));
+		captureScreenCenter(newScreenshotCallback("screenshot_center.png"));
+		return super.onTouchEvent(event);
+	}
+
+	private GVRScreenshotCallback newScreenshotCallback(final String filename) {
+		return new GVRScreenshotCallback() {
+
+			@Override
+			public void onScreenCaptured(Bitmap bitmap) {
+				if (bitmap != null) {
+					File file = new File(
+							Environment.getExternalStorageDirectory(),
+							filename);
+					try {
+						file.createNewFile();
+					} catch (IOException e) {
+						e.printStackTrace();
+					}
+					FileOutputStream outputStream = null;
+					try {
+						outputStream = new FileOutputStream(file);
```

Comment:
This code is more complex than it needs to be: The `FileOutputStream (File file)` docs say "The file will be ... created if it doesn't exist."